oauthproxy.go

```diff
@@ -810,19 +810,30 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 // addHeadersForProxying adds the appropriate headers the request / response for proxying
 func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState) {
 	if p.PassBasicAuth {
+		req.Header.Del("Authorization")
+		req.Header.Del("X-Forwarded-Email")
+		req.Header.Del("X-Forwarded-User")
+
 		req.SetBasicAuth(session.User, p.BasicAuthPassword)
 		req.Header["X-Forwarded-User"] = []string{session.User}
 		if session.Email != "" {
 			req.Header["X-Forwarded-Email"] = []string{session.Email}
 		}
 	}
 	if p.PassUserHeaders {
+		req.Header.Del("X-Forwarded-Email")
+		req.Header.Del("X-Forwarded-User")
+
 		req.Header["X-Forwarded-User"] = []string{session.User}
 		if session.Email != "" {
 			req.Header["X-Forwarded-Email"] = []string{session.Email}
 		}
 	}
 	if p.SetXAuthRequest {
+		rw.Header().Del("X-Auth-Request-Access-Token")
+		rw.Header().Del("X-Auth-Request-Email")
+		rw.Header().Del("X-Auth-Request-User")
```

Comment:
This runs before `p.serveMux.ServeHTTP(rw, req)`, so I think it is useless to try delete headers of the response writer here. If they are set by the upstream, they will happen *after* this.

Reply:
This is a very good point from @ploxiln, if we make the change to if/else statements, we can drop these Headers from the response as they _should_ at this point be empty anyway